src/transport/http-transport.js

```diff
@@ -41,17 +42,21 @@ class HTTPTransport extends Transport {
         'Content-Type': 'application/octet-stream',
       },
       body: request.data,
-    })
-      .then((response) => {
-        // Only resolve if response was successful (status of 200-299)
-        if (response.ok) { 
-          return response;
-        }
-        return Promise.reject(response); 
-      });
+    });
+
+    if (!response.ok) {
+      throw new Error(`Fetch failed with status ${response.status}`);
+    }
+
+    const web5ResponseHeader = response.headers.get(this.ENCODED_RESPONSE_HEADER);
+    if (web5ResponseHeader) {
+      // RecordsRead responses return `message` and `status` as header values, with a `data` ReadableStream in the body.
+      const { entries = null, message, record, status } = await this.decodeMessage(web5ResponseHeader);
+      return { entries, message, record: { data: response.body, ...record }, status };
```

Comment:
this means that if `record` has a `data` property it will override the `data` you specify.  id suggest putting the `...record` first so that the `data: response.body` will never be overriden

Reply:
This would not present an issue if you were using the TBD provided libraries for HTTP transport on both the provider and client app.   However, there is the potential for another implementer to accidentally include a `data` property in the `record` object, which would overwrite and cause issues.  To be defensive, I adjusted this in 78d637d1e0646c06fae6ec3b0fd7dbac930bd995